--- a/project/SquerylBuild.scala	2014-01-24 10:31:20.000000000 -0600
+++ c/project/SquerylBuild.scala	2014-01-24 13:04:20.000000000 -0600
@@ -12,17 +12,6 @@
       organization := "org.squeryl",
       version := "0.9.5-6",
       javacOptions := Seq("-source", "1.6", "-target", "1.6"),
-      version <<= version { v => //only release *if* -Drelease=true is passed to JVM
-        val release = Option(System.getProperty("release")) == Some("true")
-        if (release)
-          v
-        else {
-          val suffix = Option(System.getProperty("suffix"))
-          var i = v.indexOf('-')
-          if (i < 0) i = v.length
-          v.substring(0, i) + "-" + (suffix getOrElse "SNAPSHOT")
-        }
-      },
       parallelExecution := false,
       publishMavenStyle := true,
       scalaVersion := "2.10.0",
