diff -up scala-2.8.0.final-sources/src/msil/ch/epfl/lamp/compiler/msil/Type.java.fix3716 scala-2.8.0.final-sources/src/msil/ch/epfl/lamp/compiler/msil/Type.java
--- scala-2.8.0.final-sources/src/msil/ch/epfl/lamp/compiler/msil/Type.java.fix3716	2010-07-30 16:22:36.170084940 -0600
+++ scala-2.8.0.final-sources/src/msil/ch/epfl/lamp/compiler/msil/Type.java	2010-07-30 16:22:42.745142175 -0600
@@ -812,7 +812,7 @@ public abstract class Type extends Membe
 	if (k >= 0)
 	    cname = cname.substring(k + 1);
 	return  "[" + t.Assembly().GetName() + "]" + t +
-	    "(" + cname + "#" + Integer.toHexString(t.##) + ")";
+	    "(" + cname + "#" + Integer.toHexString(t.hashCode()) + ")";
     }
     private static String dumpType(Type t) {
 	StringBuffer str = new StringBuffer();
