diff -up scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/Completion.scala.systemjline scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/Completion.scala
--- scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/Completion.scala.systemjline	2010-10-13 22:03:52.000000000 +0200
+++ scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/Completion.scala	2010-12-09 18:59:26.431432002 +0100
@@ -309,7 +309,8 @@ class Completion(val repl: Interpreter) 
       else xs.reduceLeft(_ zip _ takeWhile (x => x._1 == x._2) map (_._1) mkString)
 
     // This is jline's entry point for completion.
-    override def complete(_buf: String, cursor: Int, candidates: JList[String]): Int = {
+    override def complete(_buf: String, cursor: Int, _candidates: JList[_]): Int = {
+      val candidates = _candidates.asInstanceOf[JList[String]]
       val buf = onull(_buf)
       verbosity = if (isConsecutiveTabs(buf, cursor)) verbosity + 1 else 0
       DBG("\ncomplete(%s, %d) last = (%s, %d), verbosity: %s".format(buf, cursor, lastBuf, lastCursor, verbosity))
diff -up scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/History.scala.systemjline scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/History.scala
--- scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/History.scala.systemjline	2010-10-13 22:03:52.000000000 +0200
+++ scala-2.8.1.final-sources/src/compiler/scala/tools/nsc/interpreter/History.scala	2010-12-09 18:52:12.258432001 +0100
@@ -14,7 +14,7 @@ import Properties.userHome
 /** Primarily, a wrapper for JLine's History.
  */
 class History(val jhistory: JHistory) {
-  def asJavaList = jhistory.getHistoryList
+  def asJavaList = jhistory.getHistoryList.asInstanceOf[java.util.List[String]]
   def asList: List[String] = asBuffer(asJavaList).toList
   def index = jhistory.getCurrentIndex
 
