diff -Nru scala-2.9.2-sources/build.xml scala-2.9.2-sources-gil/build.xml
--- scala-2.9.2-sources/build.xml	2012-04-04 18:03:13.000000000 +0200
+++ scala-2.9.2-sources-gil/build.xml	2012-09-13 02:07:24.630865166 +0200
@@ -164,7 +164,7 @@
   <!-- Sets location of pre-compiled libraries -->
   <property name="lib.starr.jar" value="${lib.dir}/scala-library.jar"/>
   <property name="comp.starr.jar" value="${lib.dir}/scala-compiler.jar"/>
-  <property name="jline.jar" value="${lib.dir}/jline.jar"/>
+  <property name="jline2.jar" value="/usr/share/java/jline2.jar"/>
   <property name="ant.jar" value="${ant.home}/lib/ant.jar"/>
   <property name="scalacheck.jar" value="${lib.dir}/scalacheck.jar"/>
 
@@ -425,7 +425,7 @@
         <pathelement location="${build-locker.dir}/classes/library"/>
         <pathelement location="${build-locker.dir}/classes/compiler"/>
         <path refid="aux.libs"/>
-        <pathelement location="${jline.jar}"/>
+        <pathelement location="${jline2.jar}"/>
       </compilationpath>
     </scalacfork>
     <propertyfile file="${build-locker.dir}/classes/compiler/compiler.properties">
@@ -644,7 +644,7 @@
         <pathelement location="${build-quick.dir}/classes/library"/>
         <pathelement location="${build-quick.dir}/classes/compiler"/>
         <path refid="aux.libs"/>
-        <pathelement location="${jline.jar}"/>
+        <pathelement location="${jline2.jar}"/>
       </compilationpath>
     </scalacfork>
     <propertyfile file="${build-quick.dir}/classes/compiler/compiler.properties">
@@ -825,7 +825,7 @@
       <pathelement location="${build-quick.dir}/classes/compiler"/>
       <pathelement location="${build-quick.dir}/classes/scalap"/>
       <path refid="aux.libs"/>
-      <pathelement location="${jline.jar}"/>
+      <pathelement location="${jline2.jar}"/>
     </path>
     <taskdef name="quick-bin" classname="scala.tools.ant.ScalaTool" classpathref="quick.bin.classpath"/>
     <mkdir dir="${build-quick.dir}/bin"/>
@@ -931,7 +931,7 @@
       <zipfileset dirmode="755" filemode="644" src="${fjbg.jar}"/>
       <zipfileset dirmode="755" filemode="644" src="${msil.jar}"/>
     </jar>
-    <copy file="${jline.jar}" toDir="${build-pack.dir}/lib"/>
+    <!--copy file="${jline.jar}" toDir="${build-pack.dir}/lib"/-->
     <copy todir="${build-pack.dir}/lib">
       <fileset dir="${lib-extra.dir}">
         <include name="**/*.jar"/>
@@ -1000,7 +1000,7 @@
       <classpath>
         <pathelement location="${build-pack.dir}/lib/scala-library.jar"/>
         <pathelement location="${build-pack.dir}/lib/scala-compiler.jar"/>
-        <pathelement location="${build-pack.dir}/lib/jline.jar"/>
+        <!--pathelement location="${build-pack.dir}/lib/jline.jar"/-->
       </classpath>
     </taskdef>
     <mkdir dir="${build-pack.dir}/bin"/>
@@ -1039,7 +1039,7 @@
       <pathelement location="${build-pack.dir}/lib/scala-partest.jar"/>
       <pathelement location="${build-pack.dir}/lib/scalap.jar"/>
       <pathelement location="${ant.jar}"/>
-      <pathelement location="${jline.jar}"/>
+      <pathelement location="${jline2.jar}"/>
       <path refid="lib.extra"/>
     </path>
     <taskdef resource="scala/tools/ant/antlib.xml" classpathref="pack.classpath"/>
@@ -1161,7 +1161,7 @@
         <pathelement location="${build-strap.dir}/classes/library"/>
         <pathelement location="${build-strap.dir}/classes/compiler"/>
         <path refid="aux.libs"/>
-        <pathelement location="${jline.jar}"/>
+        <pathelement location="${jline2.jar}"/>
       </compilationpath>
     </scalacfork>
     <propertyfile file="${build-strap.dir}/classes/compiler/compiler.properties">
diff -Nru scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/ConsoleReaderHelper.scala scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/ConsoleReaderHelper.scala
--- scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/ConsoleReaderHelper.scala	2012-03-18 17:09:24.000000000 +0100
+++ scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/ConsoleReaderHelper.scala	2012-09-13 01:58:09.000000000 +0200
@@ -6,8 +6,8 @@
 package scala.tools.nsc
 package interpreter
 
-import scala.tools.jline.console.{ ConsoleReader, CursorBuffer }
-import scala.tools.jline.console.completer.CompletionHandler
+import jline.console.{ ConsoleReader, CursorBuffer }
+import jline.console.completer.CompletionHandler
 import Completion._
 
 trait ConsoleReaderHelper extends ConsoleReader {
diff -Nru scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/Delimited.scala scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/Delimited.scala
--- scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/Delimited.scala	2012-03-18 17:09:24.000000000 +0100
+++ scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/Delimited.scala	2012-09-13 01:58:43.000000000 +0200
@@ -6,7 +6,7 @@
 package scala.tools.nsc
 package interpreter
 
-import scala.tools.jline.console.completer.ArgumentCompleter.{ ArgumentDelimiter, ArgumentList }
+import jline.console.completer.ArgumentCompleter.{ ArgumentDelimiter, ArgumentList }
 
 class JLineDelimiter extends ArgumentDelimiter {
   def toJLine(args: List[String], cursor: Int) = args match {
diff -Nru scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/JLineCompletion.scala scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/JLineCompletion.scala
--- scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/JLineCompletion.scala	2012-03-18 17:09:26.000000000 +0100
+++ scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/JLineCompletion.scala	2012-09-13 01:59:30.000000000 +0200
@@ -6,8 +6,8 @@
 package scala.tools.nsc
 package interpreter
 
-import scala.tools.jline._
-import scala.tools.jline.console.completer._
+import jline._
+import jline.console.completer._
 import Completion._
 import collection.mutable.ListBuffer
 
diff -Nru scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/JLineReader.scala scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/JLineReader.scala
--- scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/JLineReader.scala	2012-03-18 17:09:26.000000000 +0100
+++ scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/JLineReader.scala	2012-09-13 02:00:06.000000000 +0200
@@ -6,8 +6,8 @@
 package scala.tools.nsc
 package interpreter
 
-import scala.tools.jline.console.ConsoleReader
-import scala.tools.jline.console.completer._
+import jline.console.ConsoleReader
+import jline.console.completer._
 import session._
 import scala.collection.JavaConverters._
 import Completion._
diff -Nru scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/Parsed.scala scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/Parsed.scala
--- scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/Parsed.scala	2012-03-18 17:09:24.000000000 +0100
+++ scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/Parsed.scala	2012-09-13 02:00:40.000000000 +0200
@@ -6,7 +6,7 @@
 package scala.tools.nsc
 package interpreter
 
-import scala.tools.jline.console.completer.ArgumentCompleter.{ ArgumentDelimiter, ArgumentList }
+import jline.console.completer.ArgumentCompleter.{ ArgumentDelimiter, ArgumentList }
 import util.returning
 
 /** One instance of a command buffer.
diff -Nru scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/ReplProps.scala scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/ReplProps.scala
--- scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/ReplProps.scala	2012-03-18 17:09:24.000000000 +0100
+++ scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/ReplProps.scala	2012-09-13 02:01:19.000000000 +0200
@@ -11,8 +11,8 @@
 class ReplProps {
   private def bool(name: String) = BooleanProp.keyExists(name)
 
-  val jlineDebug = bool("scala.tools.jline.internal.Log.debug")
-  val jlineTrace = bool("scala.tools.jline.internal.Log.trace")
+  val jlineDebug = bool("jline.internal.Log.debug")
+  val jlineTrace = bool("jline.internal.Log.trace")
   val noThreads  = bool("scala.repl.no-threads")
 
   val info  = bool("scala.repl.info")
diff -Nru scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/session/package.scala scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/session/package.scala
--- scala-2.9.2-sources/src/compiler/scala/tools/nsc/interpreter/session/package.scala	2012-03-18 17:09:24.000000000 +0100
+++ scala-2.9.2-sources-gil/src/compiler/scala/tools/nsc/interpreter/session/package.scala	2012-09-13 02:01:54.000000000 +0200
@@ -13,10 +13,10 @@
   type JIterator[T]       = java.util.Iterator[T]
   type JListIterator[T]   = java.util.ListIterator[T]
 
-  type JEntry             = scala.tools.jline.console.history.History.Entry
-  type JHistory           = scala.tools.jline.console.history.History
-  type JMemoryHistory     = scala.tools.jline.console.history.MemoryHistory
-  type JPersistentHistory = scala.tools.jline.console.history.PersistentHistory
+  type JEntry             = jline.console.history.History.Entry
+  type JHistory           = jline.console.history.History
+  type JMemoryHistory     = jline.console.history.MemoryHistory
+  type JPersistentHistory = jline.console.history.PersistentHistory
 
   private[interpreter] implicit def charSequenceFix(x: CharSequence): String = x.toString
 }
