diff -up scala-2.9.1.final-sources/src/compiler/scala/tools/ant/templates/tool-unix.tmpl.tool scala-2.9.1.final-sources/src/compiler/scala/tools/ant/templates/tool-unix.tmpl
--- scala-2.9.1.final-sources/src/compiler/scala/tools/ant/templates/tool-unix.tmpl.tool	2011-05-02 12:20:26.000000000 +0200
+++ scala-2.9.1.final-sources/src/compiler/scala/tools/ant/templates/tool-unix.tmpl	2011-11-27 20:15:00.986325419 +0100
@@ -47,24 +47,15 @@ case "`uname`" in
     CYGWIN*) cygwin=true ;;
 esac
 
-# Finding the root folder for this Scala distribution
-SOURCE=$0;
-SCRIPT=`basename "$SOURCE"`;
-while [ -h "$SOURCE" ]; do
-    SCRIPT=`basename "$SOURCE"`;
-    LOOKUP=`ls -ld "$SOURCE"`;
-    TARGET=`expr "$LOOKUP" : '.*-> \(.*\)$'`;
-    if expr "${TARGET:-.}/" : '/.*/$' > /dev/null; then
-        SOURCE=${TARGET:-.};
-    else
-        SOURCE=`dirname "$SOURCE"`/${TARGET:-.};
-    fi;
-done;
-
-# see #2092
-SCALA_HOME=`dirname "$SOURCE"`
-SCALA_HOME=`cd "$SCALA_HOME"; pwd -P`
-SCALA_HOME=`cd "$SCALA_HOME"/..; pwd`
+export JAVA_HOME=/usr/lib/jvm/java-1.6.0/
+
+[ -r @@JAVADIR@@-utils/java-functions ] && . @@JAVADIR@@-utils/java-functions || exit 1
+
+set_javacmd || exit 3
+check_java_env || exit 4
+set_jvm_dirs || exit 5
+
+SCALA_HOME="@@DATADIR@@/scala"
 
 # Remove spaces from SCALA_HOME on windows
 if $cygwin; then
@@ -84,23 +75,6 @@ if [ -z "$TOOL_CLASSPATH" ] ; then
     done
 fi
 
-CYGWIN_JLINE_TERMINAL=
-if $cygwin; then
-    if [ "$OS" = "Windows_NT" ] && cygpath -m .>/dev/null 2>/dev/null ; then
-        format=mixed
-    else
-        format=windows
-    fi
-    SCALA_HOME=`cygpath --$format "$SCALA_HOME"`
-    TOOL_CLASSPATH=`cygpath --path --$format "$TOOL_CLASSPATH"`
-    case "$TERM" in
-        rxvt* | xterm*)
-            stty -icanon min 1 -echo
-            CYGWIN_JLINE_TERMINAL="-Djline.terminal=scala.tools.jline.UnixTerminal"
-        ;;
-    esac
-fi
-
 [ -n "$JAVA_OPTS" ] || JAVA_OPTS="@javaflags@"
 
 # break out -D and -J options and add them to JAVA_OPTS as well
@@ -146,11 +120,7 @@ done
 # reset "$@@" to the remaining args
 set -- "${scala_args[@@]}"
 
-if [ -z "$JAVACMD" -a -n "$JAVA_HOME" -a -x "$JAVA_HOME/bin/java" ]; then
-    JAVACMD="$JAVA_HOME/bin/java"
-fi
-
-"${JAVACMD:=java}" \
+exec ${JAVACMD} \
   $JAVA_OPTS \
   "${java_args[@@]}" \
   ${CPSELECT}${TOOL_CLASSPATH} \
